app/v1/controllers/requestParsers/ListBsasRequestDataParser.scala

```diff
@@ -20,11 +20,28 @@ package v1.controllers.requestParsers
 import javax.inject.Inject
 import uk.gov.hmrc.domain.Nino
 import v1.controllers.requestParsers.validators.ListBsasValidator
+import v1.controllers.requestParsers.validators.validations.TypeOfBusinessValidation
 import v1.models.request.{DesTaxYear, ListBsasRawData, ListBsasRequest}
 
 class ListBsasRequestDataParser @Inject()(val validator: ListBsasValidator)
   extends RequestParser[ListBsasRawData, ListBsasRequest] {
 
-  override protected def requestFor(data: ListBsasRawData): ListBsasRequest =
-    ListBsasRequest(Nino(data.nino), DesTaxYear.fromMtd(data.taxYear), data.typeOfBusiness, data.selfEmploymentId)
+  override protected def requestFor(data: ListBsasRawData): ListBsasRequest = {
+
+    val incomeSourceIdentifier = (data.selfEmploymentId, data.typeOfBusiness) match {
+      case (Some(_), _) => Some("incomeSourceId")
+      case (None, Some(_)) => Some("incomeSourceType")
+      case (None, None) => None
+    }
+    val identifierValue: Option[String] = if(data.selfEmploymentId.isDefined) data.selfEmploymentId else data.typeOfBusiness match {
+      case Some(TypeOfBusinessValidation.selfEmployed) => Some("01")
+      case Some(TypeOfBusinessValidation.ukPropertyFHL) => Some("04")
+      case Some(TypeOfBusinessValidation.ukPropertyNonFHL) => Some("02")
+      case _ => None
```

Comment:
The `TypeOfBusiness` enum set has the mappings to the DES representation already.

Avoid string literals knocking around in the codebase.